# Aspirin - Omega3 - cancer graph
## NCATS Hackathon 2019-09-17



How does aspirin interact with omega-3 fatty acids to reduce the risk of cancer?

<img src="img/AspirinFattyAcidsCancer.png" width="1200">

**IMPORTANT**: Current BTE limitation is that any single query is limited to 100 results per query, so it may not entirely show all results from each source

### Initiating the package

In [1]:
# import the query module
from biothings_explorer.user_query_dispatcher import SingleEdgeQueryDispatcher
# import the hint module (suggest hits based on your input)
from biothings_explorer.hint import Hint
# import the registry module
from biothings_explorer.registry import Registry
reg = Registry()
ht = Hint()
# initialize the connect module
from biothings_explorer.user_query_dispatcher import Connect

## Find the fatty acid nodes

In [57]:
fa_commonnames = [
    "Alpha-linolenic acid",
    "Eicosapentaenoic acid",
    "Docosahexaenoic acid",
    "Linoleic acid",
    "Arachidonic acid"
]


In [58]:
fa_nodes = []
for name in fa_commonnames:
    # use the hint the module to let BioThings Explorer suggest the inputs for you
    a = ht.query(name)
    # the output of the hint module is grouped by semantic types
    fa_nodes.append(a['ChemicalSubstance'][0])
    
fa_nodes

[{'chembl': 'CHEMBL8739',
  'drugbank': 'DB00132',
  'name': 'LINOLENIC ACID',
  'pubchem': 5280934,
  'display': 'chembl(CHEMBL8739) drugbank(DB00132) name(LINOLENIC ACID) pubchem(5280934) ',
  'type': 'ChemicalSubstance',
  'primary': {'identifier': 'chembl',
   'cls': 'ChemicalSubstance',
   'value': 'CHEMBL8739'}},
 {'pubchem': 6437059,
  'display': 'pubchem(6437059) ',
  'type': 'ChemicalSubstance',
  'primary': {'identifier': 'pubchem',
   'cls': 'ChemicalSubstance',
   'value': 6437059}},
 {'drugbank': 'DB05297',
  'pubchem': 6918473,
  'display': 'drugbank(DB05297) pubchem(6918473) ',
  'type': 'ChemicalSubstance',
  'primary': {'identifier': 'drugbank',
   'cls': 'ChemicalSubstance',
   'value': 'DB05297'}},
 {'chembl': 'CHEMBL267476',
  'drugbank': 'DB14104',
  'name': 'LINOLEIC ACID',
  'pubchem': 5280450,
  'display': 'chembl(CHEMBL267476) drugbank(DB14104) name(LINOLEIC ACID) pubchem(5280450) ',
  'type': 'ChemicalSubstance',
  'primary': {'identifier': 'chembl',
   'cls':

### what connected nodes are shared in common with these FAs?

In [59]:
# first test genes

all_nodes={}
for node in fa_nodes:
    try:
        seqd = SingleEdgeQueryDispatcher(input_obj=node,
                                 output_cls='Gene',
                                 registry=reg)
        seqd.query()
    except:
        print("Passing on this record: "+str(node))
        continue
    for x in seqd.show_all_nodes():
        if x in all_nodes.keys():
            all_nodes[x].append(node['primary']['value'])
        else:
            all_nodes[x] = [node['primary']['value']]
all_nodes

Passing on this record: {'pubchem': 6437059, 'display': 'pubchem(6437059) ', 'type': 'ChemicalSubstance', 'primary': {'identifier': 'pubchem', 'cls': 'ChemicalSubstance', 'value': 6437059}}


{'CHEMBL8739': ['CHEMBL8739'],
 '3992': ['CHEMBL8739'],
 '9415': ['CHEMBL8739'],
 '2864': ['CHEMBL8739', 'CHEMBL267476'],
 '338557': ['CHEMBL8739', 'CHEMBL267476'],
 '6546': ['CHEMBL8739'],
 '6785': ['CHEMBL8739'],
 '7442': ['CHEMBL8739'],
 '2170': ['CHEMBL8739', 'CHEMBL267476'],
 '1588': ['CHEMBL8739', 'CHEMBL267476'],
 '5743': ['CHEMBL8739', 'CHEMBL267476'],
 '5742': ['CHEMBL8739', 'CHEMBL267476', 'CHEMBL15594'],
 '5468': ['CHEMBL8739', 'CHEMBL267476'],
 '5465': ['CHEMBL8739', 'CHEMBL267476', 'CHEMBL15594'],
 '5467': ['CHEMBL8739', 'CHEMBL267476'],
 '2152': ['CHEMBL8739', 'CHEMBL267476'],
 '9971': ['CHEMBL8739', 'CHEMBL15594'],
 '6256': ['CHEMBL8739', 'CHEMBL15594'],
 'DB05297': ['DB05297'],
 '596': ['DB05297'],
 'CHEMBL267476': ['CHEMBL267476'],
 '3172': ['CHEMBL267476'],
 '3745': ['CHEMBL267476'],
 '79054': ['CHEMBL267476'],
 'LOX1.1': ['CHEMBL267476'],
 '134': ['CHEMBL267476'],
 '2167': ['CHEMBL267476'],
 'CHEMBL15594': ['CHEMBL15594'],
 '338567': ['CHEMBL15594'],
 '50801': ['CHEM

* `5465` is PPARA
* `5742` is PTGS1

#### What is the relationship between PPARA and those compounds?

In [76]:
node = fa_nodes[0]
seqd = SingleEdgeQueryDispatcher(input_obj=node,
                                 output_cls='Gene',
                                 registry=reg)
seqd.query()
seqd.display_edge_info('CHEMBL8739', '5465')
node

{'chembl': 'CHEMBL8739',
 'drugbank': 'DB00132',
 'name': 'LINOLENIC ACID',
 'pubchem': 5280934,
 'display': 'chembl(CHEMBL8739) drugbank(DB00132) name(LINOLENIC ACID) pubchem(5280934) ',
 'type': 'ChemicalSubstance',
 'primary': {'identifier': 'chembl',
  'cls': 'ChemicalSubstance',
  'value': 'CHEMBL8739'}}

In [78]:
for node in fa_nodes:
    try:
        seqd = SingleEdgeQueryDispatcher(input_obj=node,
                                 output_cls='Gene',
                                 registry=reg)
        seqd.query()
        print('NODE: '+node['chembl'])
        print(seqd.display_edge_info(node['chembl'], '5465'))
    except:
        continue


NODE: CHEMBL8739
{0: {'info': {'bts:name': ['Peroxisome proliferator-activated receptor alpha'], 'bts:symbol': ['PPARA'], '@type': 'Gene', '$input': 'bts:chembl', '$source': 'drugcentral'}, 'label': 'bts:target', 'source': 'drugcentral'}}
NODE: CHEMBL267476
{0: {'info': {'bts:name': ['Peroxisome proliferator-activated receptor alpha'], 'bts:symbol': ['PPARA'], '@type': 'Gene', '$input': 'bts:chembl', '$source': 'drugcentral'}, 'label': 'bts:target', 'source': 'drugcentral'}}
NODE: CHEMBL15594
{0: {'info': {'bts:name': ['Peroxisome proliferator-activated receptor alpha'], 'bts:symbol': ['PPARA'], 'bts:function': ['Zinc ion binding'], 'bts:pubmed': [10403814, 15491415], '@type': 'Gene', '$input': 'bts:drugbank', '$source': 'drugbank'}, 'label': 'bts:target', 'source': 'drugbank'}}


In [ ]:
seqd.show_all_edges()

In [42]:
# test compounds

all_nodes={}
for node in fa_nodes:
    try:
        seqd = SingleEdgeQueryDispatcher(input_obj=node,
                                 output_cls='ChemicalSubstance',
                                 registry=reg)
        seqd.query()
    except:
        print("Passing on this record: "+str(node))
        continue
    for x in seqd.show_all_nodes():
        if x in all_nodes.keys():
            all_nodes[x].append(node['primary']['value'])
        else:
            all_nodes[x] = [node['primary']['value']]
all_nodes

Passing on this record: {'pubchem': 6437059, 'display': 'pubchem(6437059) ', 'type': 'ChemicalSubstance', 'primary': {'identifier': 'pubchem', 'cls': 'ChemicalSubstance', 'value': 6437059}}


{}

Nothing on compounds -- I think this is a BTE bug... (https://github.com/kevinxin90/bte_schema/issues/7)

In [47]:
# test pathway

all_nodes={}
for node in fa_nodes:
    try:
        seqd = SingleEdgeQueryDispatcher(input_obj=node,
                                 output_cls='Pathway',
                                 registry=reg)
        seqd.query()
    except:
        print("Passing on this record: "+str(node))
        continue
    for x in seqd.show_all_nodes():
        if x in all_nodes.keys():
            all_nodes[x].append(node['primary']['value'])
        else:
            all_nodes[x] = [node['primary']['value']]
all_nodes

Passing on this record: {'pubchem': 6437059, 'display': 'pubchem(6437059) ', 'type': 'ChemicalSubstance', 'primary': {'identifier': 'pubchem', 'cls': 'ChemicalSubstance', 'value': 6437059}}


{}

In [67]:
# test disease

all_nodes={}
for node in fa_nodes:
    try:
        seqd = SingleEdgeQueryDispatcher(input_obj=node,
                                 output_cls='DiseaseOrPhenotypicFeature',
                                 registry=reg)
        seqd.query()
    except:
        print("Passing on this record: "+str(node))
        continue
    for x in seqd.show_all_nodes():
        if x in all_nodes.keys():
            all_nodes[x].append(node['primary']['value'])
        else:
            all_nodes[x] = [node['primary']['value']]
all_nodes

Passing on this record: {'pubchem': 6437059, 'display': 'pubchem(6437059) ', 'type': 'ChemicalSubstance', 'primary': {'identifier': 'pubchem', 'cls': 'ChemicalSubstance', 'value': 6437059}}


{'CHEMBL8739': ['CHEMBL8739'],
 'CHEMBL267476': ['CHEMBL267476'],
 'CHEMBL15594': ['CHEMBL15594']}

### What is link between PPARA and inflammation

In [55]:
# search for inflammation
a = ht.query("inflammatory response")
node_inflammation = a['BiologicalProcess'][0]

# search for ppara
a = ht.query("PPARA")
node_ppara = a['Gene'][0]

In [56]:
cc = Connect(input_obj=node_inflammation, output_obj=node_ppara, registry=reg)
cc.connect()

processing step 1 ...
processing step 2 ...
query completed
Connction not found!


**Ask Kevin** Is this a bug?  Should we provide more info on what was tried here?